website/django/data/searchtest.py

```diff
@@ -0,0 +1,947 @@
+            'Reference_Sequence':'HGVS_cDNA',
+            'Gene_Symbol':'Protein_Change'
+        }
+
+        #This loops through test_list dictionary, providing matching pairs of correctly-ordered search terms as field_1 and field_2
+        for field_1, field_2 in test_list.items():
+            test_case = getattr(self.existing_variant_materialized_view, field_1) + ':' + getattr(self.existing_variant_materialized_view, field_2)
+
+            request = self.factory.get(
+            '/data/?format=json&order_by=Gene_Symbol&direction=ascending&page_size=20&page_num=0&search_term=%s&include=Variant_in_ENIGMA&include=Variant_in_ClinVar&include=Variant_in_1000_Genomes&include=Variant_in_ExAC&include=Variant_in_LOVD&include=Variant_in_BIC&include=Variant_in_ESP&include=Variant_in_exLOVD' % test_case)
+            response = index(request)
+            #build a string for failure message
+            message = 'Test case for \'' + field_1 + ':' + field_2 + '\' failed!!!'
+
+            self.assertIsInstance(response, JsonResponse, message)
+            self.assertEqual(response.status_code, 200, message)
+
+            response_data = json.loads(response.content)
+
+            self.assertEqual(response_data['count'], 1, message)
+
+            response_variant = response_data['data'][0]
+            self.assertEqual(response_variant[field_2], getattr(self.existing_variant,field_2), message)
+
+
+    def test_search_by_gene_symbol_and_hgvs_protein_with_colon(self):
+        '''Tests searching for 'BRCA1 p.(Ala280Gly) --> Gene_Symbol HGVS_Protein.split(':')[1]' with colon separator
+        BRCA1:p.(Ala280Gly) --> Gene_Symbol:HGVS_Protein.split(':')[1] (HGVS_Protein is actually stored as NP_009225.1:p.(Ala280Gly), so this has to be split on the ':')'''
+        gene_symbol_hgvs_protein = self.existing_variant_materialized_view.Gene_Symbol + ':' + self.existing_variant_materialized_view.HGVS_Protein.split(':')[1]
+        request = self.factory.get(
+            '/data/?format=json&order_by=Gene_Symbol&direction=ascending&page_size=20&page_num=0&search_term=%s&include=Variant_in_ENIGMA&include=Variant_in_ClinVar&include=Variant_in_1000_Genomes&include=Variant_in_ExAC&include=Variant_in_LOVD&include=Variant_in_BIC&include=Variant_in_ESP&include=Variant_in_exLOVD' % gene_symbol_hgvs_protein)
+        response = index(request)
+
+        self.assertIsInstance(response, JsonResponse)
+        self.assertEqual(response.status_code, 200)
+
+        response_data = json.loads(response.content)
+
+        self.assertEqual(response_data['count'], 1)
+
+        response_variant = response_data['data'][0]
+        self.assertEqual(response_variant['HGVS_Protein'], self.existing_variant.HGVS_Protein)
+
+
+    def test_search_by_hgvs_protein_and_protein_change_with_colon(self):
+        '''Tests searching for 'HGVS_Protein:Protein_Change' with colon separator
+        NP_009225.1:A280G --> HGVS_Protein.split(':')[0] Protein_Change'''
```

Comment:
this should be: `NP_009225.1:A280G --> HGVS_Protein.split(':')[0]:Protein_Change'''` (notice the added colon between hgvs protein and protein change)